### DISCUSSION TITLE
Exif metadata values on different platform

### DISCUSSION THREAD

**Discussion by Achooo** [libvips/libvips#2320] [General]
Hi! I'm not sure this is a bug, or just my lack of knowledge about exif metadata with C API on different platforms, so opening a discussion topic :) 

I'm wondering about the value of `exif-ifd2-ColorSpace` and `exif-ifd2-FlashpixVersion` metadata on an image.

I am running libvips in C (compiled with Go with CGO)
I obtain the metadata value with the `vips_image_get_as_string` method.

- I get a value of `"65535 (Uncalibrated, Short, 1 components, 2 bytes)" ` **on MacOS**
- Meanwhile I obtain `"65535 (Internal error (unknown value 65535), Short, 1 components, 2 bytes)"` **on Ubuntu Focal**
- Additionally, the `exif-ifd2-FlashpixVersion` is empty in Ubuntu, but present in MacOS.

The `internal error` in the Color Space field looks a little odd to me. Is this expected behaviour?

Running exiftool on the same image:
- shows `Color Space                     : Uncalibrated` in both cases
- shows the Flash Pix Version as `Flashpix Version                : 0100` in both cases
 

Test failure on Ubuntu Focal:
```
                Error:          Not equal: 
                                expected: "65535 (Uncalibrated, Short, 1 components, 2 bytes)"
                                actual  : "65535 (Internal error (unknown value 65535), Short, 1 components, 2 bytes)"
                                
              Error:          Not equal: 
                                expected: "FlashPix Version 1.0 (FlashPix Version 1.0, Undefined, 4 components, 4 bytes)"
                                actual  : ""
```

Again everything is working fine on MacOS. 

Thank you for your help 🙏🏽

**Comment by jcupitt** [comment]
Hello @Achooo,

I expect it's because the versions of libexif are different. It looks like focal has libexif12_0.6.21-6ubuntu0.4, but homebrew has libexif-0.6.22.
